gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/handlers/DeploymentHandler.java

```diff
@@ -316,29 +351,39 @@ private void deleteCollectionNamespaceStandardPod(final String gafferNamespace,
                 });
     }
 
-    private List<GaaSGraph> listAllGraphs(final KubernetesClient kubernetesClient, final List<String> apiDeployments) {
+    private List<GaaSGraph> listAllGraphs(final KubernetesClient kubernetesClient, final List<String> graphIds) {
         List<GaaSGraph> graphs = new ArrayList<>();
-        for (final String gaffer : apiDeployments) {
+        for (final String graphId : graphIds) {
             try {
                 GaaSGraph gaaSGraph = new GaaSGraph();
-                gaaSGraph.graphId(gaffer);
-                Collection<String> graphConfig = kubernetesClient.configMaps().inNamespace(NAMESPACE).withName(gaffer + "-gaffer-graph-config").get().getData().values();
+                gaaSGraph.graphId(graphId);
+                Collection<String> graphConfig = kubernetesClient.configMaps().inNamespace(NAMESPACE).withName(graphId + "-gaffer-graph-config").get().getData().values();
                 gaaSGraph.description(getValueOfConfig(graphConfig, "description"));
                 if (getValueOfConfig(graphConfig, "configName") != null) {
                     gaaSGraph.configName(getValueOfConfig(graphConfig, "configName"));
                 }
-                int availableReplicas = kubernetesClient.apps().deployments().inNamespace(NAMESPACE).withName(gaffer + GAFFER_NAME_SUFFIX).get().getStatus().getAvailableReplicas();
+                int availableReplicas = kubernetesClient.apps().deployments().inNamespace(NAMESPACE).withName(graphId + GAFFER_NAME_SUFFIX).get().getStatus().getAvailableReplicas();
                 if (availableReplicas >= 1) {
                     gaaSGraph.status(RestApiStatus.UP);
                 } else {
                     gaaSGraph.status(RestApiStatus.DOWN);
                 }
-                gaaSGraph.url("http://" + gaffer + "-" + NAMESPACE + "." + INGRESS_SUFFIX + "/ui");
-                gaaSGraph.restUrl("https://" + gaffer + "-" + NAMESPACE + "." + INGRESS_SUFFIX + "/rest");
+                gaaSGraph.url("http://" + graphId + "-" + NAMESPACE + "." + INGRESS_SUFFIX + "/ui");
+                gaaSGraph.restUrl("https://" + graphId + "-" + NAMESPACE + "." + INGRESS_SUFFIX + "/rest");
+
+                List<Deployment> deploymentList = kubernetesClient.apps().deployments().inNamespace(NAMESPACE).withLabel("app.kubernetes.io/instance", graphId).list().getItems();
+                if (!deploymentList.isEmpty()) {
+                    for (final Deployment deployment : deploymentList) {
+                        if (deployment.getMetadata().getLabels().get("graphAutoDestroyDate") != null) {
+                            gaaSGraph.graphAutoDestroyDate(deployment.getMetadata().getLabels().get("graphAutoDestroyDate").toLowerCase().replaceAll("_", ":"));
+                        }
+                        gaaSGraph.graphAutoDestroyDate("");
```

Comment:
What will happen if empty?

Reply:
When graphLifetimeInDays is empty or not exist  then graphAutoDestroyDate label will not be created.
and graph will never be auto destroyed.